src/components/ApplyDiscount.js

```diff
@@ -0,0 +1,83 @@
+import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import constants from '../../src/constants';
+import _ from 'ramda';
+
+class ApplyDiscount extends Component {
+  constructor(props) {
+    super(props);
+    this.renderDiscounts = this.renderDiscounts.bind(this);
+    this.handleAddDiscount = this.handleAddDiscount.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.state = {
+      discounts: this.props.appliedDiscounts
+    };
+  }
+
+  handleAddDiscount(discount) {
+    const { _id } = discount;
+    const idx = _.findIndex(_.propEq('_id', _id))(this.state.discounts);
+    const updatedDiscounts = idx > -1 ?
+      _.remove(idx, 1)(this.state.discounts) :
```

Comment:
Removing it to toggle selecting/deselecting, and am using state.discounts to display what's selected. does that make sense? maybe the naming is confusing...?